runtime/src/tasks/qthreads/tasks-qthreads.c

```diff
@@ -892,18 +892,21 @@ void chpl_task_addToTaskList(chpl_fn_int_t       fid,
     chpl_bool serial_state = chpl_task_getSerial();
     chpl_fn_p requested_fn = chpl_ftable[fid];
 
-    assert(subloc != c_sublocid_none);
+    assert(full_subloc != c_sublocid_none);
 
     PROFILE_INCR(profile_task_addToTaskList,1);
 
     if (serial_state) {
         // call the function directly.
         requested_fn(arg);
     } else {
+        c_sublocid_t execution_subloc =
+          chpl_localeModel_sublocToExecutionSubloc(full_subloc);
```

Comment:
I'm curious if this is going to result in any noticeable increase in task-spawning times for flat. I doubt it, but it's something to keep an eye out for.